get-cloudify.py

```diff
@@ -535,100 +552,276 @@ def check_cloudify_installed(virtualenv_path=None):
+    # undocumented, this should only be a problem when someone is switching
+    # from old to new args and puts different values for each.
+    deprecations = {
+        'withrequirements': 'with_requirements',
+        'installvirtualenv': 'install_virtualenv',
+        'installpip': 'install_pip',
+        'pythonpath': 'python_path',
+        'wheelspath': 'wheels_path',
+        'forceonline': 'force_online',
+        'installpycrypto': 'install_pycrypto',
+        'installpythondev': 'install_pythondev',
+    }
+    for deprecated, new_value in deprecations.items():
+        if new_value in parsed_args and parsed_args[deprecated]:
+            logger.warning(
+                '--{depvar} is deprecated. Use --{newvar}. '
+                '--{depvar} will be removed in a future release.'.format(
+                    depvar=deprecated.replace('_', '-'),
+                    newvar=new_value.replace('_', '-'),
+                )
+            )
+            # Now make sure we use the value that was set
+            parsed_args[new_value] = parsed_args[deprecated]
+
+    # Process branch selection, if applicable
+    if parsed_args['use_branch']:
+        user = 'cloudify-cosmo'
+        branch = parsed_args['use_branch']
+        if '/' in branch:
+            if len(branch.split('/')) > 2:
+                parser.error('--use-branch should be specified either as '
+                             '<branch> or as <user>/<branch>. '
+                             'Too many "/" found in arguments.')
+            else:
+                user, branch = parsed_args['use_branch'].split('/')
+
+        # Form source string from user and branch
+        url = repo_url.format(user=user, branch=branch)
 
+        parsed_args['source'] = url
 
-lgr = init_logger(__file__)
+    # use_branch should be discarded now as it is just used to set source
+    parsed_args.pop('use_branch')
+
+    if parsed_args['source'] and not parsed_args['with_requirements']:
+        logger.warning(
+            'A source URL or branch was specified, but '
+            '--with-requirements was omitted. You may need to retry using '
+            '--with-requirements if the installation fails.'
+        )
+
+    return parsed_args, deprecations.keys()
+
+
+logger = init_logger(__file__)
 
 
 if __name__ == '__main__':
-    args = parse_args()
-    if args.quiet:
-        lgr.setLevel(logging.ERROR)
-    elif args.verbose:
-        lgr.setLevel(logging.DEBUG)
+    args, deprecated = parse_args()
+    if args['quiet']:
+        logger.setLevel(logging.ERROR)
+    elif args['verbose']:
+        logger.setLevel(logging.DEBUG)
     else:
-        lgr.setLevel(logging.INFO)
-    handle_upgrade(args.upgrade, args.virtualenv)
-
-    xargs = ['quiet', 'verbose']
-    args = {arg: v for arg, v in vars(args).items() if arg not in xargs}
+        logger.setLevel(logging.INFO)
+    handle_upgrade(args['upgrade'], args['virtualenv'])
```

Comment:
I'd move handle_upgrade to the class. I have no idea why I put it here in the first place.